ci/jenkins/Jenkins_steps.groovy

```diff
@@ -1604,6 +1604,27 @@ def docs_prepare() {
 }
 
 
+def docs_prepare_beta() {
+    return ['Prepare for publication of the full website': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.init_git()
+
+            unstash 'jekyll-artifacts'
+            unstash 'python-artifacts'
+
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_docs', false)
+
+            // archive so the publish pipeline can access the artifact
+            archiveArtifacts 'docs/_build/beta_website.tgz'
```

Comment:
where is this being prepared ?

Reply:
Did I miss something?
Job calls ci/jenkins/Jenkinsfile_website_beta
 --> unix_cpu_lite --> binary
 --> docs_jekyll --> jekyll artifacts (stash)
 --> docs_python --> python api artifacts (stash)
         --> docs_prepare_beta (unstash both) --> archive beta_website.tgz
                --> docs_publish_beta (using archive)

I did forget to update the filename in the publish job, so this was a worthwhile exercise  🏃 😄

Reply:
You are running `utils.docker_run('ubuntu_cpu_jekyll', 'build_docs', false)`
check runtime_function.sh:1819, the `doc/_build/beta_website.tgz` is not going to be created by magic 😄 
You need to create a new runtime function, `build_docs_beta` for example that would do something like this:

```
build_docs_beta() {
    pushd docs/_build
    tar -xzf jekyll-artifacts.tgz
    api_folder='html/api'

    # Python has it's own landing page/site so we don't put it in /docs/api
    mkdir -p $api_folder/python/docs && tar -xzf python-artifacts.tgz --directory $api_folder/python/docs
    GZIP=-9 tar -zcvf beta_website.tgz -C html .
    popd
}
```

Reply:
But please let's try to not diverge between bets and prod code as much as possible.

Reply:
> But please let's try to not diverge between bets and prod code as much as possible.

This does have the drawback of not rendering most of the other microsites, but then we get a much faster way to stage updates. How about after most of the restructuring is done on the Python API we go back and add in the rest of the APIs so the staging run has everything in it?

Reply:
Sounds good! :)